link-area-delegation-explainer.md

```diff
@@ -0,0 +1,141 @@
+
+### Link delegation attributes
+Expressing link area delegation in HTML can look something like this:
+```html
+<section class="card" linkarea>
+  <a href="/post?id=123" defaultlink>Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
+```
+
+In the above example, the `section` element has a `linkarea` attribute, which would delegate its click to the first descendant with a `defaultlink` attribute, in this case the first `a` element.
+
+We could have different similar alternatives to this with different ergonomic tradeoffs, e.g. using ID references:
+```html
+<section class="card" link="card123-title">
+  <a href="/post?id=123" id="card123-title">Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
+```
+
+Note that in this example, the link still has to be a descendant of the card. This is to guide web authors to an interactive structure where the links are in the appropriate place in the tree.
+
+## Alternatives considered
+
+### A new element type
+One of the alternatives that sound natural is a new element type:
+```html
+<section class="card">
+  <linkarea>
+    <a href="/post?id=123" defaultlink>Post Title</a>
+    <img>
+    <button>Join</button>
+    <button>Share</button>
+  </linkarea>
+</section>
+```
+
+This can feel compelling in a semantic sense, but becomes undesirable when we think of the flexibility of attributes.
+For example, a use case for link area delegation is making table rows clickable.
+
+```html
+<table>
+  <tr linkarea>
+     <td>1.</td>
+     <td><a href="/details?id=1" defaultlink>Title</a>
+  </tr>
+</table>
+```
+
+This is not possible with a new element, as `tr` (and some other container types) have special HTML parser behavior, and allowing them to include new types of elements would
+break the page in browsers that don't support the new element type.
+
+### CSS instead of HTML
+
+Since this proposal focuses on click areas, perhaps CSS is better suited to this than HTML?
+Something like:
+
+```html
+<section class="card">
+  <a href="/post?id=123">Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
+```
+
+```css
+.card { pointer-area: contain; }
+.card a:nth-child(1) { pointer-area: expand; }
+```
+
+This has nice ergonomics because the selection of link area can be done outside of the main structure of the page using CSS selectors.
+However, something about this feels limiting, as it is constrained to click areas in advance, and doesn't lend itself to future UA enhancements of this experience, such as context menus.
+
+### Use invokers
+One interesting alternative from @jaffathecake is to use [invokers](https://open-ui.org/components/invokers.explainer/) for this, e.g.:
+```html
+<section class="card" commandfor="card123-title" command="click">
+  <a href="/post?id=123" id="card123-title">Post Title</a>
+  <img>
+  <button>Join</button>
+  <button>Share</button>
+</section>
```

Comment:
I love this idea! I would make this small tweak:

```suggestion
<section class="card" commandfor="card123-title" commands="click contextmenu">
  <a href="/post?id=123" id="card123-title">Post Title</a>
  <img>
  <button>Join</button>
  <button>Share</button>
</section>
```

Reply:
I think it would be better to make the default command "auto", which is some sensible activation default. This could be toggle for dialog/popover, and default link activation for links. 

Right click behaviour is a good callout though. Also things like command click / middle click.